src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala

```diff
@@ -105,11 +107,33 @@ class ScalafixInterface private (
       try arg(acc)
       catch { case NonFatal(e) => throw new InvalidArgument(e.getMessage) }
     }
-    new ScalafixInterface(newScalafixArguments, this.args ++ args)
+    new ScalafixInterface(
+      newScalafixArguments,
+      postScalafixHooks,
+      this.args ++ args
+    )
   }
 
   def run(): Seq[ScalafixError] =
-    scalafixArguments.run().toSeq
+    try {
+      scalafixArguments.run().toSeq
+    } finally {
+      val mayRewrite = args.reverse
+        .collectFirst {
+          case Arg.ParsedArgs(args) => args
+        }
+        .forall { args =>
+          Seq("--check", "--test", "--auto-suppress-linter-errors")
+            .intersect(args)
+            .isEmpty
+        }
```

Comment:
that's the only solid heuristic I could think of to reduce the number of unnecessary callbacks